x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -587,6 +603,13 @@ private LogicalPlan resolveEnrich(Enrich enrich, List<Attribute> childrenOutput)
             }
             return enrich;
         }
+
+        private static final DataType[] GEO_TYPES = new DataType[] { GEO_POINT, GEO_SHAPE };
+        private static final DataType[] NON_GEO_TYPES = new DataType[] { KEYWORD, IP, LONG, INTEGER, FLOAT, DOUBLE, DATETIME };
```

Comment:
I think `TEXT` can be used for `match` and `range` types.

Reply:
I'm thinking of refining the edge cases in followup PRs, since this one has become quite large.

Reply:
++